### PR TITLE
[dagit] Replace opName with opNames everywhere

### PR DESCRIPTION
**Pull request by hellendag**
### Summary & Motivation

Replace Asset `opName` with `opNames` throughout Dagit.

### How I Tested These Changes

TS, lint, jest.

I'm not sure how to test these specifically within Dagit, so I could use some help sanity checking.

### CODE REVIEW

**Comment by hellendag** [comment]
Current dependencies on/for this PR:
* master
  * **PR #7977** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/7977" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/7977?utm_source=stack-comment).

**Review by prha — APPROVED** [review]
yeah, this all makes sense, I think?

File: js_modules/dagit/packages/core/src/asset-graph/AssetNode.tsx [hunk 1]
```diff
@@ -31,7 +31,8 @@ export const AssetNode: React.FC<{
   padded?: boolean;
   inAssetCatalog?: boolean;
 }> = React.memo(({definition, selected, liveData, inAssetCatalog, padded = true}) => {
-  const stepKey = definition.opName || '';
+  const firstOp = definition.opNames.length ? definition.opNames[0] : null;
+  const stepKey = firstOp || '';
```
Comment (bengotow) [inline js_modules/dagit/packages/core/src/asset-graph/AssetNode.tsx:35]: Hmm I'm not sure where this was being used, but i think we want to pass all the opNames anywhere we were passing stepKey

**Review by bengotow — APPROVED** [review]
This looks good to me! I'm pretty confident in these changes except the one place I flagged we may want to check  🙌

File: js_modules/dagit/packages/core/src/asset-graph/AssetNode.tsx (hunk 1 above)
Comment (hellendag) [reply js_modules/dagit/packages/core/src/asset-graph/AssetNode.tsx:35]: This is for highlighting a step on the Run page filtered logs. I don't think we currently support filtering to more than one step.

Comment (hellendag) [reply js_modules/dagit/packages/core/src/asset-graph/AssetNode.tsx:38]: Added a comment here.

**Pull request merged**